volume/volume.go

```diff
@@ -0,0 +1,24 @@
+package volume
+
+type Driver interface {
+	// Name returns the name of the volume driver.
+	Name() string
+	// Create makes a new volume with the given id.
+	Create(string) (Volume, error)
+	// Remove deletes the volume.
+	Remove(Volume) error
+}
+
+type Volume interface {
```

Comment:
It seems like most of the implementation for this is boiler plate that will need to be duplicated across each driver.